### PR TITLE
Add hasUnpartitionedCookieAccess

### PR DESCRIPTION
**Pull request by arichiv**
### Description

This is the first of 3 PRs adding spec changes from https://privacycg.github.io/saa-non-cookie-storage/ launching in Chrome 125.

### Motivation

hasUnpartitionedCookieAccess is a renaming of hasStorageAccess intended to provide more clarity around what storage is actually accessible. This is important as Chrome now supports requesting unpartitioned storage access with or without unpartitioned cookies.

### Additional details

https://chromestatus.com/feature/5175585823522816

### Related issues and pull requests

Relates to https://github.com/mdn/mdn/issues/543

### CODE REVIEW

**Comment by arichiv** [comment]
@Elchi3 @pepelsbey Have time to review?

**Comment by arichiv** [comment]
@sideshowbarker I'm working on https://github.com/mdn/browser-compat-data/pull/22978 but that's paused until the new spec I added to browser-specs propagates. Can this more forward before then?

**Comment by sideshowbarker** [comment]
> I'm working on [mdn/browser-compat-data#22978](https://github.com/mdn/browser-compat-data/pull/22978) but that's paused until the new spec I added to browser-specs propagates. Can this more forward before then?

I think the project policy is to not merge mdn/content PRs that for features that need data that’s not yet in mdn/browser-compat-data (BCD) — the reason being that unless/until the data is in BCD, the presentation of the MDN Browser Compat table and Specifications section of the corresponding MDN page will be broken.

Specifically in this case, it seems like `api.Document.hasUnpartitionedCookieAccess` needs to be merged into main in BCD before this PR can be landed.

And if the BCD PR for `api.Document.hasUnpartitionedCookieAccess` is blocked on waiting for a new browser-specs to be deployed, I also don’t know myself how it takes for that to happen

**Comment by arichiv** [comment]
@Elchi3 @pepelsbey this is ready for review, the only flaw is that it's dependent on pulling two versions of packages newer than what's in the lockfile.

**Review by pepelsbey — COMMENTED** [review]
Loog good in general, just a few suggestions :)

File: files/en-us/web/api/document/hasstorageaccess/index.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ The **`hasStorageAccess()`** method of the {{domxref("Document")}} interface ret
 
 This method is part of the [Storage Access API](/en-US/docs/Web/API/Storage_Access_API).
 
+This method is another name for {{DOMxRef("Document.hasUnpartitionedCookieAccess()")}}.
+
+There are no current plans to remove this method in favor of {{DOMxRef("Document.hasUnpartitionedCookieAccess()")}}.
```
Comment (pepelsbey) [inline files/en-us/web/api/document/hasstorageaccess/index.md:17]:
```suggestion
> **Note:** This method is another name for {{DOMxRef("Document.hasUnpartitionedCookieAccess()")}}. There are no current plans to remove this method in favor of {{DOMxRef("Document.hasUnpartitionedCookieAccess()")}}.
```

I’d say this part stands out from the surrounding content as extra info or a note. What do you think?

**Review by pepelsbey — APPROVED** [review]
Looks great! Thank you :) Feel free to accept the suggestion if you agree.

**Comment by arichiv** [comment]
I don't have permission to merge on this or the other one, could you do so?

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**